tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +48,71 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new ApiWebResponse((HttpWebResponse)exception.Response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            var boundary = "---------------------------" + DateTime.Now.Ticks.ToString("x");
+
+            _request.Method = "POST";
+            _request.ContentType = "multipart/form-data; boundary=" + boundary;
+
+            using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                // Write form request using the boundary
+
+                var boundaryBytes = Encoding.ASCII.GetBytes("\r\n--" + boundary + "\r\n");
+
+                // Write each MultipartFormItem
+                foreach (var item in items)
+                {
+                    await requestStream.WriteAsync(boundaryBytes, 0, boundaryBytes.Length).ConfigureAwait(false);
+                    byte[] headerBytes;
+                    if (item.FileName is null)
+                    {
+                        headerBytes = Encoding.UTF8.GetBytes(
+                            $"Content-Disposition: form-data; name=\"{item.Name}\"\r\nContent-Type: {item.ContentType}\r\n\r\n");
```

Comment:
Should we handle `"` in `item.Name` and `item.FileName`? Not sure if they're allowed spec-wise, if not we could _probably_ just throw, or we could replace the invalid char etc. There's options 🤷‍♂️ . Or we could ignore it, as this (presumably) will only be used by us

Reply:
Validation added in [84ecdc7](https://github.com/DataDog/dd-trace-dotnet/pull/2962/commits/84ecdc7a6638c4593e83777456e9370644539de4)